### ISSUE TITLE
Feature Request for Clamp

### ISSUE THREAD

**Issue by Stubs42** [Coirt/Bark#23]
Hi,

Sry if this is not the right place for a feature request.

would it be possible to implement two more switches like the link switch with the following functionality ?:

1. A clamp/modulo switch:
Currently the input is just clamped which would be the default.
When modulo is switched on, the input is not just clamped but using a modulo operation to calculate the output. So a input of 3.5V for a 0V min to 2V max would result in a 1.5V output.
Something like out = min + ((in - min) % (max - min))
Didn't think about negative Voltages (can be solved by just adding a constant like 10V to all in/min/max, doing the calculation and subtracting the constant for output) but the idea should be clear.

2, A keep note switch on/off
Defaults to 'off'
When set to yes, clamp should try to clamp the input in a way that the note (in % 1) is the same for the output. So a C will still be a C (maybe in another octave) on output.
This is possible if max-min >= 1, otherwise its not always possible to do that and it will ignore this switch for input notes not between existing between min and max.
When modulo switch is off it will output the highest same note if in > max and lowest same note if in < min otherwise out will be the nearest same note in the min/max range after the modulo calculation.

With those features clamp could be very handy to keep melodies or bass lines in a certain pitch range without destroying the harmonies. A typical example is that your bass line has notes too low or too high to sound good which should be transposed to an octave where they sound good.
In other words, I would like to use clamp as an dynamic transposer to keep notes in a defined well sounding pitch range. 

Regards
Dieter

**Comment by Coirt** [comment]
Hi Dieter, the clamp is not applied to the incoming signal, it is applied to the output.

The signal chain works like this, input > attenuverter > gain > clamp > output.

Let me have a think about it and I'll get back to you.

**Comment by Stubs42** [comment]
Hi Coirt,
Ok, Clamp seems not  not to be designed to do what I'm looking for.
Since I'm a software developer too, I fully understand that new functionality may conflict with original design having this functionality not in mind.
Maybe a new module fulfilling my wishes will come up some time.
Thanks for your answer
Regards
Dieter

**Comment by Coirt** [comment]
The issue I see with using modulas is the scale of the output. It would be okay for audio as the min/max voltages are the amplitude but for v/oct it would rescale/attenuate the signal which would be out of key. Considering 1v = 1 octave, 1/12 = 0.083v * 6 semitones = 0.4999999v. The 1.5v would be out of key. I believe it would require a new module to function correctly.

Would a quantizer work currently for this functionality?
